### PR TITLE
Object store/ azure testing

### PR DESCRIPTION
**Pull request by deepsidhu85**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Added testing for azure object storage. All methods in the `IridaFileStorageAzureUtilityImpl.java` are tested against a dockerized mock of an azure storage blob.

To run the tests locally you need to install `Azurite` on your machine, create a directory where the blobs and debug file will be stored and start up the service in a terminal using the command `azurite --location /path/to/blobs --debug /path/to/debugfile` where the `/path/to` is the path to the directory

No other modifications are necessary for testing. See https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azurite

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
* [X] Tests added (or description of how to test) for any new features.
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

**Review by ericenns — CHANGES_REQUESTED** [review]
In general this is looking good, just a few comments below. Also could you update `run-tests.sh` to start the docker container for the corresponding tests that require it.

File: .github/workflows/ci-test.yml [hunk 1]
```diff
@@ -39,7 +39,13 @@ jobs:
             "galaxy_pipeline_testing",
             "open_api_testing",
           ]
-
+    services:
+      azurite:
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply .github/workflows/ci-test.yml]: The service entry doesn't allow an `if` statement. Instead I added the docker stuff for Azurite in the `steps` section. It now only starts up when running the FileSystem tests.

File: src/test/java/ca/corefacility/bioinformatics/irida/ria/unit/cloud/IridaFileStorageAzureUtilityTest.java [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import com.azure.storage.common.StorageSharedKeyCredential;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class IridaFileStorageAzureUtilityTest implements IridaFileStorageTestUtility {
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/test/java/ca/corefacility/bioinformatics/irida/ria/unit/cloud/IridaFileStorageAzureUtilityTest.java]: Converted it over to an IntegrationTest. I decided to create a new tag for the FileSystem as this allows us to just start up the azurite docker container for these specific tests. If we reused the `Service` tag then we would have azurite starting up for those tests as well.

**Review by ericenns — APPROVED** [review]
👍 I have tested this locally and it works for me, thanks!

**Pull request merged**